tools/chapel-py/src/chapel/visitor/__init__.py

```diff
@@ -0,0 +1,90 @@
+import chapel.core
+
+from collections import defaultdict
+from inspect import signature
+
+def _try_call(dispatch_table, visitor, node, method_type):
+    node_type = type(node)
+    while node_type is not object:
+        if node_type in dispatch_table:
+            methods = dispatch_table[node_type]
```

Comment:
Can you add a comment or two explaining the logic of this loop? Particularly the python inheritance walking